### ISSUE TITLE
Security.conf & Wordpress.conf is blocking access to WP backend!

### ISSUE THREAD

**Issue by PontusTideman** [digitalocean/nginxconfig.io#182]
#103  Information

## Details
*6523 access forbidden by rule
*6586 access forbidden by rule
*5 access forbidden by rule
*1 access forbidden by rule

### Description 
```
# security headers
add_header X-Frame-Options           "SAMEORIGIN" always;
add_header X-XSS-Protection          "1; mode=block" always;
add_header X-Content-Type-Options    "nosniff" always;
add_header Referrer-Policy           "strict-origin-when-cross-origin" always;
add_header Content-Security-Policy   "default-src 'self' https: data: 'unsafe-inline' 'unsafe-eval';" always;
add_header Strict-Transport-Security "max-age=31536000; includeSubDomains" always;

# . files
location ~ /\.(?!well-known) {
    deny all;
}
```

Is blocking access to the WP backend. When commenting out the security.conf from the sites .conf it started working.

### Steps to reproduce
Create a new conf as above!

### Expected behavior
To work and not block?

**Comment by PontusTideman** [comment]
Oh, and the above is also throwing
```
session expired
```
On a brand new WP site. After I have commented out the security.conf that also went away!

**Comment by PontusTideman** [comment]
Update:
Also wordpress.conf is blocking admin access to Media tab (e.g upload.php).

**Comment by MattIPv4** [comment]
What dotfiles are you/Wordpress trying to access that security.conf is blocking?

**Comment by PontusTideman** [comment]
The whole backend is just white with texts... 
<img width="618" alt="Screenshot 2020-10-16 at 14 32 04" src="https://user-images.githubusercontent.com/9256550/96258571-7ec46300-0fbc-11eb-93b7-acc6a0399a46.png">

When I comment out security and wordpress the backend returns and works as it should. Have allowed my IP in the settings with no success also! O_o

**Comment by PontusTideman** [comment]
And the error.log is full with warnings for my IP trying to access:

```
*4363 access forbidden by rule, client: 193.180.18.13, server: HIDDEN.TLD, request: "GET /wp-admin/load-scripts.php?c=0&load%5Bchunk_0%5D=jquery-core,utils&ver=5.5.1 HTTP/2.0", host: "HIDDEN.TLD", referrer: "https://HIDDEN.TLD/wp-admin/"
```

**Comment by MattIPv4** [comment]
```nginx
# WordPress: deny scripts and styles concat
location ~* \/wp-admin\/load-(?:scripts|styles)\.php {
    deny all;
}
```

I'm guessing this is the rule at fault then -- if you comment out just that rule, do things function as expected?
(These rules pre-date me maintaining this project, but will dig through the history to figure out when/why this was added)

**Comment by PontusTideman** [comment]
Yes, that did the trick!
Would be fun to know WHY deny the backend was implemented for... O_o

**Comment by MattIPv4** [comment]
Looks like this rule was originally added in https://github.com/digitalocean/nginxconfig.io/commit/62bd8f3e6be3d24d017f5d13365588717c7a4439

Researching the CVE, it seems this isn't really a suggested fix for the issue, so I think the best solution may just be to remove this rule from the config completely.

**Comment by PontusTideman** [comment]
That CVE is for up to 4.9.3 version of WP! A version that was released 2,5 years back. ^^

**Comment by ihorvorotnov** [comment]
@PontusTideman I've done some research on this and checked the actual changes to WordPress core code. 4.9.3 *does not fix* anything even remotely related to this CVE, although many 3rd party sources mention this. See all changes here - https://core.trac.wordpress.org/query?status=closed&group=resolution&milestone=4.9.3.

As far as I understand, this still hasn't been fixed. Here's the entire history of this file: https://github.com/WordPress/WordPress/commits/master/wp-admin/load-scripts.php

However, completely blocking access to these files is not a solution, and never was. These files are crucial for the WordPress admin to run. You can rate-limit requests, implement partial blocking (allow yourself via IP whitelist etc) but never block them completely.

**Comment by MattIPv4** [comment]
:nod: I feel this isn't something that nginxconfig should try to fix, there are WP plugins etc. that can help with mitigating the vulnerability.

I'll make a PR to remove this block rule 👍
